test/ReverseProxy.Tests/Service/Config/ProxyRouteTransformExtensionsTests.cs

```diff
@@ -0,0 +1,369 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.AspNetCore.Routing.Template;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Net.Http.Headers;
+using Microsoft.ReverseProxy.Abstractions;
+using Microsoft.ReverseProxy.Common;
+using Microsoft.ReverseProxy.Service.RuntimeModel.Transforms;
+using Xunit;
+
+namespace Microsoft.ReverseProxy.Service.Config
+{
+    public class ProxyRouteTransformExtensionsTests
+    {
+        [Fact]
+        public void AddTransformPathSet()
+        {
+            var proxyRoute = CreateProxyRoute();
+
+            proxyRoute.AddTransformPathSet(new PathString("/path#"));
+
+            var transform = BuildTransform(proxyRoute);
+
+            var requestTransform = Assert.Single(transform.RequestTransforms);
+            var pathStringTransform = Assert.IsType<PathStringTransform>(requestTransform);
+            Assert.Equal(PathStringTransform.PathTransformMode.Set, pathStringTransform.Mode);
+            Assert.Equal("/path%23", pathStringTransform.Value);
```

Comment:
This shouldn't have changed formats.
```suggestion
            Assert.Equal("/path#", pathStringTransform.Value.Value);
```

Same for the rest of these tests.